raiden/ui/cli.py

```diff
@@ -686,25 +689,33 @@ def run_app(
 
     blockchain_service = BlockChainService(privatekey_bin, rpc_client)
 
-    net_id = blockchain_service.network_id
-    if net_id != network_id:
-        if network_id in ID_TO_NETWORKNAME and net_id in ID_TO_NETWORKNAME:
+    given_numeric_network_id = network_id.value if isinstance(network_id, ChainId) else network_id
+    node_numeric_network_id = blockchain_service.network_id
+    known_given_network_id = networkid_is_known(given_numeric_network_id)
+    known_node_network_id = networkid_is_known(node_numeric_network_id)
+    if known_given_network_id:
+        given_network_id = ChainId(given_numeric_network_id)
+    if known_node_network_id:
+        node_network_id = ChainId(node_numeric_network_id)
+
+    if node_numeric_network_id != given_numeric_network_id:
+        if known_given_network_id and known_node_network_id:
             click.secho(
-                f"The chosen ethereum network '{ID_TO_NETWORKNAME[network_id]}' "
-                f"differs from the ethereum client '{ID_TO_NETWORKNAME[net_id]}'. "
+                f"The chosen ethereum network '{ID_TO_NETWORKNAME[given_network_id]}' "
+                f"differs from the ethereum client '{ID_TO_NETWORKNAME[node_network_id]}'. "
```

Comment:
Nitpicks: The `ID_TO_NETWORKNAME` lookups aren't necessary. Since the value is an enum we can just use `given_network_id.name.lower()`